src/components/ui/button/Button.vue

```diff
@@ -24,71 +25,40 @@ export interface Props {
 
 const props = withDefaults(defineProps<Props>(), {
 	spinnerBorder: 2,
-	type: 'button'
+	type: 'button',
+	size: 'md'
 })
 
-let componentButton = 'BUTTON'
-const classList = ref<string[]>([])
-const trailingIcon = ref(props.trailingIcon)
-
-if (props.to) {
-	componentButton = 'router-link'
-}
-
-if (props.href) {
-	componentButton = 'a'
-}
-
-if (props.size) {
-	classList.value.push(`-${props.size}`)
-}
-
-if (props.variant) {
-	classList.value.push(`-${props.variant}`)
-} else {
-	classList.value.push(`-default`)
-}
-
-if (props.leadingIcon) {
-	classList.value.push(`-icon`)
-}
-
-if (props.flush) {
-	classList.value.push(`-flush-${props.flush}`)
-}
-
-if (props.block) {
-	classList.value.push(`-block`)
-}
-
-if (props.outline) {
-	classList.value.push(`-outline`)
-}
-
-if (props.disclosure) {
-	trailingIcon.value = 'arrow_drop_down'
-	classList.value.push('-disclousure')
-}
+const componentType = computed(() => {
+	if (props.to) return 'router-link'
+	if (props.href) return 'a'
+	return 'BUTTON'
+})
 
-watch(
-	() => props.variant,
-	(newVal) => {
-		classList.value.push(`-${newVal}`)
-	}
-)
+const classList = computed(() => {
+	const filteredClassList = [
```

Comment:
```suggestion
return [
  props.size ? `-${props.size}` : '',
  props.variant ? `-${props.variant}` : '-default',
  props.leadingIcon ? '-icon' : '',
  props.flush ? `-flush-${props.flush}` : '',
  props.block ? '-block' : '',
  props.outline ? '-outline' : '',
  props.disclosure ? '-disclosure' : ''
]
```
O que acha dessa solução? Assim evitamos o filter para os `undefined` e facilitamos a leitura